packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -793,6 +807,7 @@ class _LargeTitleNavigationBarSliverDelegate
   @override
   Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
     final bool showLargeTitle = shrinkOffset < maxExtent - minExtent - _kNavBarShowLargeTitleThreshold;
+    final double expandedState = shrinkOffset > minExtent ? clampDouble((shrinkOffset - minExtent) / (maxExtent - minExtent) * 2, 0, 1) : 0;
```

Comment:
I'd suggest a different calculation for `expandedState` so it looks closer as native:
```dart
final double largeTitleThreshold = maxExtent - minExtent - _kNavBarShowLargeTitleThreshold;
final bool showLargeTitle = shrinkOffset < largeTitleThreshold;
final double expandedState = shrinkOffset > largeTitleThreshold + _kNavBarShowLargeTitleThreshold
   ? clampDouble((shrinkOffset - largeTitleThreshold) / largeTitleThreshold * 2, 0, 1)
    : 0;
```